docs/markdown/star_formation.md

```diff
@@ -1,29 +1,54 @@
+## Jeans instability filter
 
+Eligible cells are first identified through a Jeans-length check before any stochastic sampling occurs.
 
-## Estimating number and type of stars
+- Compute the Jeans length \(\lambda_J = c_s / \sqrt{G \rho}\) in every cell.
+- Mark the cell as eligible when \(\lambda_J < J \cdot \Delta x\) with \(J = 0.5\).
+- Only eligible cells continue to the sampling steps below.
 
-Once we establish that a star will form in a cell, we need to find out the stellar population looks like. In our implementation, the population comprises at least one particle representing the low mass star ($M<8M_{\odot}$) population plus a random number of particles, each representing a high mass star.
+## Controlling the formation rate
 
-The number of high mass stars is determined by the initial mass function, taken to be Chabrier03. The IMF is represented by a log normal for $M<M_{\odot}$ and a power law with a slope of $2.35$ above that. From the IMF, we can estimate the mass fraction of the high mass stars, $f_{*,\rm{high}}$, and the average mass of high mass stars, $\langle m \rangle _{*,\rm{high}}$. The expectation value of the number of high mass stars, *num_high_mass_stars*, then is the ratio of the total mass in high mass stars and $\langle m \rangle _{*,\rm{high}}$. 
+Two efficiency parameters tune how aggressively eligible gas is converted into star particles during each hydro step.
 
-The total number of star particles to be spawned in a cell is $1+$ Poission distribution with an expectation value of *num_high_mass_stars*. A fraction $1-f_{*,\rm{high}}$ goes into the particle representing low mass stars while the mass of the high mass star particles is randomly drawn from the high mass part of the IMF.
+- \(\epsilon_{ff}\): efficiency per free-fall time, defined through \(\epsilon_{ff} = (\dot M_{\star} \, t_{ff}) / (M_{cell} \, \Delta t)\).
+- \(\epsilon_{\star}\): fraction of the cell mass used when a particle is spawned.
+- The target stellar mass for the step is \(\epsilon_{ff} M_{cell} (\Delta t / t_{ff})\).
+- Bernoulli probability for spawning: \( P = \frac{\epsilon_{ff}}{\epsilon_{\star}} \frac{\Delta t}{t_{ff}} \).
+- The expectation value \(\langle M_{\star} \rangle = P \epsilon_{\star} M_{cell}\) matches the target mass provided \(\Delta t < t_{ff}\); the CFL condition typically enforces that inequality.
 
-In the case there are no high mass stars in the cell, the low mass star particle gets spawned at the centre of the cell with a velocity identical to the gas velocity. 
+## Sampling the stellar population
 
+Once a cell passes the filter and the Bernoulli draw succeeds, we construct the composite stellar population represented by the spawned particles.
 
+- Every accepted draw creates one low-mass particle that represents all stars with \(M < 8 M_{\odot}\).
+- High-mass stars follow the Chabrier (2003) IMF: log-normal below \(1 M_{\odot}\) and a slope of \(2.35\) above it.
```

Comment:
I think what we implemented is the Chabrier (2005) IMF, which is slightly different: http://adsabs.harvard.edu/abs/2005ASSL..327...41C